tests/test_number.py

```diff
@@ -0,0 +1,169 @@
+"""Test the frigate number."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from pytest_homeassistant_custom_component.common import async_fire_mqtt_message
+
+from homeassistant.const import ATTR_ENTITY_ID
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+
+from . import (
+    TEST_CONFIG_ENTRY_ID,
+    TEST_NUMBER_FRONT_DOOR_CONTOUR_AREA_ENTITY_ID,
+    TEST_NUMBER_FRONT_DOOR_THRESHOLD_ENTITY_ID,
+    create_mock_frigate_client,
+    enable_and_load_entity,
+    setup_mock_frigate_config_entry,
+    test_entities_are_setup_correctly_in_registry,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+DISABLED_NUMBER_ENTITY_IDS = {
+    TEST_NUMBER_FRONT_DOOR_CONTOUR_AREA_ENTITY_ID,
+    TEST_NUMBER_FRONT_DOOR_THRESHOLD_ENTITY_ID,
+}
+
+
+async def test_number_state(hass: HomeAssistant) -> None:
+    """Verify a successful number setup."""
+    client = create_mock_frigate_client()
+    await setup_mock_frigate_config_entry(hass, client=client)
+
+    for entity_id in DISABLED_NUMBER_ENTITY_IDS:
+        entity_state = hass.states.get(entity_id)
+        assert not entity_state
+        await enable_and_load_entity(hass, client, entity_id)
+
+    async_fire_mqtt_message(hass, "frigate/available", "online")
+    await hass.async_block_till_done()
+
+    for entity_id in DISABLED_NUMBER_ENTITY_IDS:
+        entity_state = hass.states.get(entity_id)
+        assert entity_state
+        assert entity_state.state is not None
+
+    async_fire_mqtt_message(hass, "frigate/front_door/motion_contour_area/state", "50")
+    await hass.async_block_till_done()
+    entity_state = hass.states.get(TEST_NUMBER_FRONT_DOOR_CONTOUR_AREA_ENTITY_ID)
+    assert entity_state
+    assert entity_state.state == "50.0"
+
+    async_fire_mqtt_message(hass, "frigate/front_door/motion_threshold/state", "255")
+    await hass.async_block_till_done()
+    entity_state = hass.states.get(TEST_NUMBER_FRONT_DOOR_THRESHOLD_ENTITY_ID)
+    assert entity_state
+    assert entity_state.state == "255.0"
+
+    async_fire_mqtt_message(hass, "frigate/available", "offline")
+    entity_state = hass.states.get(TEST_NUMBER_FRONT_DOOR_CONTOUR_AREA_ENTITY_ID)
+    assert entity_state
+    assert entity_state.state == "unavailable"
+
+
+async def test_bad_numbers(hass: HomeAssistant) -> None:
+    """Verify bad state."""
+    client = create_mock_frigate_client()
+    await setup_mock_frigate_config_entry(hass, client=client)
+
+    for entity_id in DISABLED_NUMBER_ENTITY_IDS:
+        await enable_and_load_entity(hass, client, entity_id)
+
+    async_fire_mqtt_message(hass, "frigate/available", "online")
+    await hass.async_block_till_done()
+
+    for entity_id in DISABLED_NUMBER_ENTITY_IDS:
+        entity_state = hass.states.get(entity_id)
+        assert entity_state
+        assert entity_state.state is not None
```

Comment:
Can we assert what it is instead?

Reply:
I am doing both so seemed a bit awkward to try to pull the state out or create a separate list of the states to check?